### PR TITLE
Align to spec v1.0.0

### PR DESCRIPTION
**Pull request by terencechain**
**What goes in here?**
Items that are not backwards compatible with current `master` and `medalla` testnet

**Notable components that should be there**
- Consensus params changes
- ~~BLSv4~~
- ~~Disv5~~

Change list:
* Mainnet rewards and penalty params
* New beacon state length for eth1 data voting
* New mainnet and minimal spec tests
* Toledo config
* Pyrmont config

### CODE REVIEW

File: beacon-chain/core/state/benchmarks_test.go [hunk 1]
```diff
@@ -16,19 +16,6 @@ import (
 
 var runAmount = 25
 
-func TestExecuteStateTransition_FullBlock(t *testing.T) {
```
Comment (terencechain) [inline beacon-chain/core/state/benchmarks_test.go:19]: I'm removing this test. This is actually more housekeeping work than good. Whenever there's a change to a beacon object, this test requires a manual update

**Review by rauljordan — APPROVED** [review]

File: shared/params/mainnet_config.go [hunk 2]
```diff
@@ -75,12 +75,12 @@ var mainnetBeaconConfig = &BeaconChainConfig{
...
-	EpochsPerEth1VotingPeriod:        32,
+	EpochsPerEth1VotingPeriod:        64,
 	SlotsPerHistoricalRoot:           8192,
 	MinValidatorWithdrawabilityDelay: 256,
 	ShardCommitteePeriod:             256,
 	MinEpochsToInactivityPenalty:     4,
-	Eth1FollowDistance:               1024,
+	Eth1FollowDistance:               2048,
 	SafeSlotsToUpdateJustified:       8,
 	SecondsPerETH1Block:              13,
```
Comment (prestonvanloon) [inline shared/params/mainnet_config.go:85]: This doesn't match spec. https://github.com/ethereum/eth2.0-specs/blob/579da6d2dc734b269dbf67aa1004b54bb9449784/configs/mainnet/phase0.yaml#L48.

In reality, eth1 block times are 13 seconds on average and Prysm's eth1 connection sends too many requests this value is wrong. Thanks to @nisdas for investigating.  Discord discussion [link](https://discord.com/channels/476244492043812875/483017808658169866/776609050757431318)

Comment (prestonvanloon) [reply shared/params/mainnet_config.go:85]: I've asked @nisdas to file an issue so we can outline our spec config deviation and potentially have the spec correct this value.

**Review by prestonvanloon — APPROVED** [review]

File: shared/params/mainnet_config.go (hunk 2 above)
Comment (terencechain) [reply shared/params/mainnet_config.go:85]: It's also worth noting that current master has been at 13s, This `v1.0.0` branch did not make this particular change.
With that said, this was the only field that did not align with latest config

https://github.com/prysmaticlabs/prysm/blob/master/shared/params/mainnet_config.go#L85

Comment (prestonvanloon) [reply shared/params/mainnet_config.go:85]: https://github.com/ethereum/eth2.0-specs/issues/2132

**Pull request merged**